code/modules/events/spacevine.dm

```diff
@@ -188,6 +188,10 @@
 /datum/spacevine_mutation/fire_proof/process_temperature(obj/effect/spacevine/holder, temp, volume)
 	return 1
 
+/datum/spacevine_mutation/fire_proof/on_hit(obj/effect/spacevine/holder, mob/hitter, obj/item/I, expected_damage)
+	if(I && I.damtype == "fire")
+		. = 0
```

Comment:
I thought welding tool was killing vines efficiently because temperature exposure, thats why fireproof mutation only had temperature exposure override